enhancements/hypershift/topology-aware-scheduling/topology-aware-scheduling.md

```diff
@@ -0,0 +1,265 @@
+Similar to: https://github.com/openshift/hypershift/pull/4150
+The KubeletConfig's ConfigMap NamespacedName convention, created by the NodePool controller is:
+`<control-plane-namespace>/<name+node-pool>`
+* `control-plane-namespace` = name of the control-plane namespace
+* `name` = name of the ConfigMap as provided by the user in the hosted-cluster namespace.
+* `node-pool` = name of the NodePool on which the ConfigMap was referenced.
+
+To make sure the final name is valid,
+it's given to the https://github.com/openshift/hypershift/blob/ffa435f71e1a037174e29d31a2b80d569c940d57/support/util/route.go#L39 
+function so that 
+```
+finalName := ShortenName(name, NodePoolName)
+```
+In addition, the NodePool controller creates the KubeletConfig ConfigMap with the following labels:
+* `hypershift.openshift.io/kubeletconfig-config` : `""`
+* `hypershift.openshift.io/nodePool` : `NodePool` API name where the KubeletConfig has referenced
+
+Then, HCCO picks it up from the control-plane namespace, 
+by listing ConfigMaps with the labels above, and copies it to the hosted-cluster.
+
+The KubeletConfig's ConfigMap NamespacedName convention, created by HCCO is:
+`openshift-config-managed/<name>`
+* `openshift-config-managed` = name of the namespace where all the configs managed by OCP reside.
+* `name` = name of the ConfigMap as provided by the NodePool controller in the control-plane namespace.
+
+Any changes to the KubeletConfig's ConfigMap on the hosted cluster will be blocked by the  
+[hypershift validating admission policy](https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/)
+IOW only HCCO can change the KubeletConfig's ConfigMap and no external tampering is allowed.
+
+When the KubeletConfig ConfigMap is deleted by the user (removed from the NodePool spec), 
+the NodePool controller deletes the ConfigMap from the control-plane namespace.
+HCCO will watch for changes in ConfigMaps under the CP NS:
+```
+p := predicate.NewPredicateFuncs(func(o client.Object) bool {
+		cm := o.(*corev1.ConfigMap)
+		if _, ok := cm.Labels[nodepool.KubeletConfigConfigMapLabel]; ok {
+			return true
+		}
+		return false
+	})
+	if err := c.Watch(source.Kind[client.Object](opts.CPCluster.GetCache(), &corev1.ConfigMap{}, eventHandler(), p)); err != nil {
+		return fmt.Errorf("failed to watch ConfigMap: %w", err)
+	}
+```
+so the deletion of the ConfigMaps triggers the reconciliation loop of HCCO.
+HCCO compares the KubeletConfig ConfigMaps on the CP NS against the KubeletConfig ConfigMaps on the
+hosted cluster, and deletes the ones exist on the hosted cluster but are no longer on the CP NS. 
+
+It is possible to have multiple KubeletConfig's ConfigMap (one per each NodePool) under the same hosted cluster.
+The names are always different because they are derived from the name provided by the user. 
+
+In case more than a single KubeletConfig's ConfigMap referenced for the NodePool,
+HCCO throws an error,
```

Comment:
This validation needs to be done at the nodepool controller level and be presented as a condition to the API consumer. Before it gets to the HCCO.

Reply:
can we have a section for this an any valdiation needed and how it'd be presented in the API conditions?

Reply:
> This validation needs to be done at the nodepool controller level 
WFM

> and be presented as a condition to the API consumer. Before it gets to the HCCO.

What do you mean API condition? 


nodepool controller also need to make sure there are no generated `KubeletConfig` which are already associated with the nodepool.

For example NTO generates a `KubeletConfig` as well when it given a `PerformanceProfile`